pkg/remotewrite/remotewrite.go

```diff
@@ -80,11 +89,51 @@ func (o *Output) Start() error {
 
 func (o *Output) Stop() error {
 	o.logger.Debug("Stopping the output")
+	defer o.logger.Debug("Output stopped")
 	o.periodicFlusher.Stop()
-	o.logger.Debug("Output stopped")
+
+	staleMarkers := o.staleMarkers(time.Now())
+	o.logger.WithField("staleMarkers", len(staleMarkers)).Debug("Marking time series as stale, if any")
+	if len(staleMarkers) < 1 {
+		return nil
+	}
+
+	// TODO: warn if it the request is taking too long (5 seconds?, the same timeout set for flushing)
+
+	err := o.client.Store(context.Background(), staleMarkers)
```

Comment:
Hhmm shouldn't this use some extension-wide context? Preferably one passed from k6? I now see that `output.Params` doesn't have a context. Maybe it should?

Reply:
>For this version, I think the unique way we have is to implement the TODO directly now logging in case of a long Stop execution. WDYT?

Thinking better on this, the current client is already initialized with the Timeout value from the configuration, so the request timed out when the timeout interval is passed. It should be enough for warning in the case of a long-time request and to alarm for a required investigation into eventual bad performance between k6 and the Remote write endpoint.

Reply:
>Intuitively, it makes sense to me for the extension context to be based on one received by k6, so that the extension code can cancel any operations in progress if the k6 context is cancelled (due to a signal, test abort, etc.).

Just to be clear, I agree here, and this is something we want to add in the future, this is why the context is passed to the `wait` function in the [Output refactor proposal](https://github.com/grafana/k6/issues/2430#issuecomment-1073803076) `(wait func(context.Context) error, err error)`.

Unfortunately, I think we can't do it now, because it would require some changes to k6 when the code is almost frozen.